### PR TITLE
Fixed clear-app-data exception

### PR DESCRIPTION
**Pull request by IlyaBausovAkvelon**
In this fix:
- Updated idb version to 1.1.7 and fixed issues after that.
- Fixed uncatched clear-app-data exception with that.
- Added Integration and Unit tests to cover clear-app-data command.

### CODE REVIEW

**Comment by AlexeyBudynkov** [comment]
At this moment these command doesn't work on M1:
```
$ make unit-tests
$ make integration-tests
```

_Testing failed:
	The linked and embedded framework 'CocoaLumberjack.framework' is missing one or more architectures required by this target: arm64.
	The linked and embedded framework 'OCMock.framework' is missing one or more architectures required by this target: arm64.
	The linked and embedded framework 'Specta.framework' is missing one or more architectures required by this target: arm64.
	The linked and embedded framework 'Expecta.framework' is missing one or more architectures required by this target: arm64.
	The linked and embedded framework 'CocoaLumberjack.framework' is missing one or more architectures required by this target: arm64.
	Testing cancelled because the build failed._

This command creates other error:
`$ make cli-tests`

_An error occurred while loading ./spec/upload_xctestconfig_spec.rb.
Failure/Error: if RunLoop::Version.new(xcode_version) >= min_xcode_version_

_NoMethodError:
  undefined method `strip' for nil:NilClass_

File: iOSDeviceManager/Devices/PhysicalDevice.m [hunk 1]
```diff
@@ -357,10 +357,12 @@ - (iOSReturnStatusCode)isInstalled:(NSString *)bundleID {
     BOOL installed = [self isInstalled:bundleID withError:&error];
 
     if (error) {
-        ConsoleWriteErr(@"Error checking if %@ is installed to %@: %@", bundleID, [self uuid], error);
-        @throw [NSException exceptionWithName:@"IsInstalledAppException"
-                                       reason:@"Unable to determine if application is installed"
-                                     userInfo:nil];
+        if ([[error localizedDescription] containsString:@"is not installed."]) {
```
Comment (AlexeyBudynkov) [inline iOSDeviceManager/Devices/PhysicalDevice.m]: What if we'll have [error localizedDescription] result in other language than English?

File: iOSDeviceManager/Utilities/DeviceUtils.m [hunk 2]
```diff
@@ -150,7 +150,7 @@ + (NSString *)defaultSimulatorID {
 
 + (NSString *)defaultSimulator {
     NSUInteger major = XcodeUtils.versionMajor + 2;
-    NSUInteger minor = XcodeUtils.versionMinor;
+    NSUInteger minor = XcodeUtils.versionMinor+1;
```
Comment (AlexeyBudynkov) [inline iOSDeviceManager/Utilities/DeviceUtils.m]: I'm not sure about this workaround.

File: IDMLogger.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+//
```
Comment (MatkovIvan) [inline IDMLogger.h]: I do not think that it should be placed at the root of repo

File: IDMLogger.m [hunk 4]
```diff
@@ -0,0 +1,230 @@
+//
```
Comment (MatkovIvan) [inline IDMLogger.m]: Same

File: IDMLogger.m [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+    return [dateFormatter stringFromDate:date];
+  }
+}
+
+- (NSString *)formatLogMessage:(DDLogMessage *)logMessage {
+  NSString *logLevel;
+  switch (logMessage.flag) {
+    case DDLogFlagError    : logLevel = @"ERROR"; break;
+    case DDLogFlagWarning  : logLevel = @" WARN"; break;
```
Comment (MatkovIvan) [inline IDMLogger.m]: Extra space in string

File: iOSDeviceManager/Devices/Simulator.m [hunk 6]
```diff
@@ -159,14 +160,18 @@ - (BOOL)waitForSimulatorState:(FBiOSTargetState)state
...
         if (error) {
             ConsoleWriteErr(@"%@", [error localizedDescription]);
         }
         return NO;
     }
 }
 
+- (FBiOSTargetState)state {
+    return self.fbSimulator.state;
+};
```
Comment (MatkovIvan) [inline iOSDeviceManager/Devices/Simulator.m]:
```suggestion
}
```

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: iOSDeviceManager/Devices/PhysicalDevice.m (hunk 1 above)
Comment (IlyaBausovAkvelon) [reply iOSDeviceManager/Devices/PhysicalDevice.m]: Fixed.

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**